### PR TITLE
feat: add support for cors and integrate waku-frontend

### PR DESCRIPTION
**Pull request by weboko**

### CODE REVIEW

**Comment by weboko** [comment]
@gabrielmer do we need to use custom nwaku image or latest to have the fix for cors?

File: run_node.sh [hunk 1]
```diff
@@ -86,6 +86,7 @@ exec /usr/bin/wakunode\
   --rest=true\
   --rest-address=0.0.0.0\
   --rest-port=8645\
+  --rest-allow-origin:"localhost:*"\
```
Comment (alrevuelta) [inline run_node.sh:89]: mm, is this flag an nwaku flag? can't find it.

**Review by alrevuelta — CHANGES_REQUESTED** [review]
thanks for the PR. left a comment.

**Comment by gabrielmer** [comment]
> @gabrielmer do we need to use custom nwaku image or latest to have the fix for cors?

To have the fix for CORS we currently need the custom image `wakuorg/nwaku:debug-cors-wa` until fixes get merged to both nim-presto and nwaku.

If necessary I can create an image with a more meaningful/appropiate name

File: run_node.sh (hunk 1 above)
Comment (gabrielmer) [reply run_node.sh:89]: This flag is going to be introduced and is not yet merged to master. We need to first merge a PR in nim-presto in order to create the PR in nwaku that would introduce this flag.

I built a custom image that includes the changes in nwaku and in nim-presto so we can use in the meantime, and in that image that flag is defined.

So if we want to use that flag, we need to also modify the image used by nwaku.

**Comment by weboko** [comment]
This PR should be ready now.

**Review by gabrielmer — APPROVED** [review]
LGTM until we get the CORS fix into master. @alrevuelta what do you think?

File: docker-compose.yml [hunk 2]
```diff
@@ -64,17 +64,14 @@ services:
...
-  #  ports:
-  #    - 127.0.0.1:4000:3000
-  #  restart: on-failure:5
-  #  depends_on:
-  #    - nwaku
+  waku-frontend:
+   # TODO: migrate to waku-org
+   image: docker.io/alrevuelta/waku-frontend:latest
+   ports:
+     - 127.0.0.1:8080:80
```
Comment (alrevuelta) [inline docker-compose.yml:71]: if the frontend listens on 3000, shouldnt this be
127.0.0.1:8080:3000


> Accepting connections at http://localhost:3000

Comment (weboko) [reply docker-compose.yml:71]: It listens `8080`

**Comment by weboko** [comment]
> I would prefer to wait until we have the cors issue fixed in a release, unless this blocks something else.

Fine by me, just to clarify - Chat is not working without it

**Comment by Ivansete-status** [comment]
@weboko confirmed that we can close that one. Thanks for it :raised_hands: !

**Pull request closed without merging**